vendor/src/github.com/docker/libnetwork/sandbox_store.go

```diff
@@ -132,7 +134,16 @@ func (sb *sandbox) storeUpdate() error {
 		sbs.Eps = append(sbs.Eps, eps)
 	}
 
-	return sb.controller.updateToStore(sbs)
+	err := sb.controller.updateToStore(sbs)
+	if err == datastore.ErrKeyModified {
+		// When we get ErrKeyModified it is sufficient to just
+		// go back and retry.  No need to get the object from
+		// the store because we always regenerate the store
+		// state from in memory sandbox state
+		goto retry
```

Comment:
It cannot be an infinite loop. If there are two racing go routines who end up here, one of them will regenerate the index properly so the other which goes through a retry will use the correct index in the next retry.